src/core_modules/capture-core/components/Pages/MainPage/TeiWorkingLists/epics/teiViewEpics/lib/getTeiListData/getTeisWithSubvalues.js

```diff
@@ -0,0 +1,159 @@
+// @flow
+import log from 'loglevel';
+import { createSelector } from 'reselect';
+import { errorCreator } from 'capture-core-utils';
+import { dataElementTypes } from '../../../../../../../../metaData';
+import type { TeiColumnsMetaForDataFetchingArray, ClientTeis } from './types';
+import type { SubvalueKeysByType, SubvaluesByType, GetTeisWithSubvaluesPlainInner } from './getTeisWithSubvalues.types';
+import type { QuerySingleResource } from '../../../../../../../../utils/api';
+
+const getTeisWithSubvaluesPlain = (querySingleResource: QuerySingleResource, absoluteApiPath: string) => {
+    const getImageOrFileResourceSubvalue = async (keys: Array<string>) => {
+        const promises = keys
+            .map(async (key) => {
+                const { id, displayName: name } = await querySingleResource({ resource: 'fileResources', id: key });
+                return {
+                    id,
+                    name,
+                };
+            });
+
+        return (await Promise.all(promises))
+            .reduce((acc, { id, name }) => {
+                acc[id] = name;
+                return acc;
+            }, {});
+    };
+
+    const subvalueGetterByType: {|[string]: any |} = {
+        [dataElementTypes.ORGANISATION_UNIT]: async (keys: Array<string>) => {
+            const ids = keys
+                .join(',');
+
+            const { organisationUnits = [] } = await querySingleResource({ resource: 'organisationUnits', params: { filter: `id:in:[${ids}]` } });
+
+            return organisationUnits
+                .reduce((acc, { id, displayName: name }) => {
+                    acc[id] = {
+                        id,
+                        name,
+                    };
+                    return acc;
+                }, {});
+        },
```

Comment:
Maybe would be better to extract this to its own function. 

Please feel free to resolve this comment in case you prefer it as it is :)

Reply:
Well, I extracted it since it took me 10 seconds, but I feel it was fine as it was.